PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/KustoQueriesWithTimeIntervalInQueryStringCheck.java

```diff
@@ -0,0 +1,255 @@
+         *
+         * @param polyadicExpression - the polyadic expression to check
+         */
+        private void handlePolyadicExpression(PsiPolyadicExpression polyadicExpression) {
+            // Process the polyadic expression to replace the variables with their values
+            PsiExpression initializer = polyadicExpression instanceof PsiPolyadicExpression ? polyadicExpression : null;
+            checkExpression(initializer, polyadicExpression);
+        }
+
+        /**
+         * This method handles the method call by checking the parameters of the method call
+         * If the parameter is a reference to a variable, the method checks the variable name
+         * If the variable name is in the list of time interval parameters, the method checks if the method call is an Azure client method call
+         * If the method call is an Azure client method call, the method registers a problem
+         *
+         * @param methodCall - the method call to check
+         */
+        private void checkParameterNames(PsiMethodCallExpressionImpl methodCall) {
+            // check the parameters of the method call
+            PsiExpressionList argumentList = methodCall.getArgumentList();
+            PsiExpression[] arguments = argumentList.getExpressions();
+
+            // for each argument in the method call, check if the argument is a reference to a variable
+            for (PsiExpression argument : arguments) {
+                if (!(argument instanceof PsiReferenceExpression)) {
+                    continue;
+                }
+                PsiReferenceExpression referenceExpression = (PsiReferenceExpression) argument;
+                PsiElement resolvedElement = referenceExpression.resolve();
+
+                if (!(resolvedElement instanceof PsiVariable)) {
+                    continue;
+                }
+                PsiVariable variable = (PsiVariable) resolvedElement;
+                String variableName = variable.getName();
+                // check if the variable name is in the list of time interval parameters
+                // if the variable name is in the list, check if the method call is an Azure client method call
+                if (!(timeIntervalParameters.contains(variableName))) {
+                    continue;
+                }
+
+                if (isAzureClient(methodCall)) {
+                    holder.registerProblem(methodCall, RULE_CONFIG.getAntiPatternMessage());
+                }
+            }
+        }
+
+
+        /**
+         * This method checks if the method call is an Azure client method call
+         * by checking the containing class of the method call
+         *
+         * @param methodCall - the method call to check
+         * @return boolean - true if the method call is an Azure client method call, false otherwise
+         */
+        private boolean isAzureClient(PsiMethodCallExpression methodCall) {
```

Comment:
Seems like we need to use this condition in several rules. We should consider making this a common utility method that all rules can use.

Reply:
I considered this but the process of checking if its a client is always relatively different depending on what I'm working on so I opted to keep the utility RuleConfig.AZURE_PACKAGE_NAME since the content of the isAzureClient method calls are usually different